plugins/elanfp/fu-elanfp-file-control.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2021
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "fu-elanfp-file-control.h"
+
+#define SIZE_IDENTIFY_PACKET 16
+
+#define S2F_TAG_FIRMWAREVERSION 0x00
+#define S2F_TAG_CFU_OFFER_A	0x72
+#define S2F_TAG_CFU_OFFER_B	0x73
+#define S2F_TAG_CFU_PAYLOAD_A	0x74
+#define S2F_TAG_CFU_PAYLOAD_B	0x75
+#define S2F_TAG_END_OF_INDEX	0xFF
+
+gboolean
+binaryVerify(const guint8 *pbinary, gsize binary_size, S2FFILE *ps2ffile, GError **error)
+{
+	S2FINDEX *ps2findex = NULL;
+	guint8 *pindex = NULL;
+	gboolean result = TRUE;
+
+	if (pbinary == NULL)
+		return FALSE;
```

Comment:
set GError, but not required if you subclass FuFirmware

Reply:
Fixed in 331c4c3c91fb57511500e163d4596d23da26e1b2